storage/innobase/dict/dict0dict.cc

```diff
@@ -4659,10 +4661,11 @@ dict_foreign_parse_drop_constraints(
 	(*constraints_to_drop)[*n] = id;
 	(*n)++;
 
-	if (std::find_if(table->foreign_set.begin(),
-			 table->foreign_set.end(),
-			 dict_foreign_matches_id(id))
-	    == table->foreign_set.end()) {
+	if (!is_exists
+	    && std::find_if(table->foreign_set.begin(),
```

Comment:
It would be safer to do `(*constraints_to_drop)[(*n)++] = id` only if the constraint was found in `table->foreign_set`. I see that `row_delete_constraint()` does not really care if the constraint exists, but we should keep in mind that constraint names of different tables may conflict. That is, the foreign key constraint name would be `database_name/constraint_name`. If you do `ALTER TABLE database_name.table_name DROP FOREIGN KEY IF EXISTS constraint_name`, with the current patch it looks like it could actually drop a constraint by that name even if it belongs to `database_name.other_table_name`.